test/integration/relations.test.js

```diff
@@ -401,123 +494,261 @@ describe('Relations', function () {
 
-                        // NOTE: make sure to count in posts with no tags
-                        //       do not count in posts with multiple tags containing one of the excluded tags
+            it('tags.slug NOT equal "classic" and posts_tags.sort_order is 0 and featured is true', function () {
+                const mongoJSON = {
+                    $and: [
+                        {
+                            $and: [
+                                {
+                                    'tags.slug': {
+                                        $ne: 'classic'
+                                    }
+                                },
+                                {
+                                    'posts_tags.sort_order': 0
+                                }
+                            ]
+                        },
+                        {
+                            featured: true
+                        }
+                    ]
+                };
+
+                const query = makeQuery(mongoJSON);
+
+                return query
+                    .select()
+                    .then((result) => {
+                        // TODO: should this include tags with no tags? the filter is about primary tag != 'classic' so they should count?
                         result.should.be.an.Array().with.lengthOf(3);
+                        result[0].title.should.equal('When She Loved Me');
+                        result[1].title.should.equal('no tags, yeah');
+                        result[2].title.should.equal('has internal tag');
+                    });
+            });
 
-                        // Check we get the right data
-                        // result.should.do.something;
+            it('tags.slug equals "animal" and posts_tags.sort_order is 0 OR author_id is 1', function () {
+                const mongoJSON = {
+                    $or: [
+                        {
+                            $and: [
+                                {
+                                    'tags.slug': 'animal'
+                                },
+                                {
+                                    'posts_tags.sort_order': 0
+                                },
+                                {
+                                    featured: false
+                                }
+                            ]
+                        },
+                        {
+                            author_id: 1
+                        }
+                    ]
+                };
+
+                const query = makeQuery(mongoJSON);
+
+                return query
+                    .select()
+                    .then((result) => {
+                        result.should.be.an.Array().with.lengthOf(6);
                     });
             });
+        });
+    });
 
-            it('can match array $nin (multiple values)', function () {
+    describe.skip('One-to-Many: Extended Cases', function () {
```

Comment:
This suite along with 'Many-to-Many: Extended Cases' was moved to the down here for better readability.